algorithms/src/polycommit/kzg10/data_structures.rs

```diff
@@ -65,33 +66,78 @@ pub struct UniversalParams<E: PairingEngine> {
     pub prepared_beta_h: <E::G2Affine as PairingCurve>::Prepared,
 }
 
+impl<E: PairingEngine> CanonicalSerialize for UniversalParams<E> {
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<(), SerializationError> {
+        self.powers.read().serialize(writer)?;
```

Comment:
Can we not just copy from `self.powers` to `writer`? This would avoid a serialize <-> deserialize round trip

Reply:
What do you mean? In case you're talking about the `read` call, it's for holding a read lock

Reply:
Ah I see. For some reason I thought the read call was reading from disk